src/main/java/ch/njol/skript/expressions/ExprBed.java

```diff
@@ -30,46 +32,47 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Bed")
-@Description("The bed location of a player, i.e. the spawn point of a player if he ever slept in a bed and the bed still exists and is unobstructed.")
-@Examples({"bed of player exists:",
-		"	teleport player the the player's bed",
-		"else:",
-		"	teleport the player to the world's spawn point"})
-@Since("2.0")
-public class ExprBed extends SimplePropertyExpression<Player, Location> {
+@Description("The bed location of a player, " +
+	"i.e. the spawn point of a player if they ever slept in a bed and the bed still exists and is unobstructed.")
+@Examples({
+	"if bed of player exists:",
+		"\tteleport player the the player's bed",
+	"else:",
+		"\tteleport the player to the world's spawn point"
+})
+@Since("2.0, INSERT VERSION (offlineplayers)")
+public class ExprBed extends SimplePropertyExpression<OfflinePlayer, Location> {
+
 	static {
-		register(ExprBed.class, Location.class, "bed[s] [location[s]]", "players");
+		register(ExprBed.class, Location.class, "bed[s] [location[s]]", "offlineplayers");
 	}
-	
+
 	@Override
 	@Nullable
-	public Location convert(final Player p) {
+	public Location convert(OfflinePlayer p) {
 		return p.getBedSpawnLocation();
 	}
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
+			if (!Player.class.isAssignableFrom(getExpr().getReturnType())) {
+				Skript.error("Bed location of offline players cannot be set/deleted.");
+				return null;
+			}
 			return new Class[] {Location.class};
+		}
 		return null;
 	}
 	
 	@Override
-	public void change(final Event e, @Nullable final Object[] delta, final ChangeMode mode) {
-		if (delta == null) {
-			for (final Player p : getExpr().getArray(e)) {
-				p.setBedSpawnLocation(null, true);
-			}
-		} else {
-			final Location l = (Location) delta[0];
-			for (final Player p : getExpr().getArray(e)) {
-				p.setBedSpawnLocation(l, true);
-			}
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		Location loc = delta == null ? null : (Location) delta[0];
+		for (OfflinePlayer p : getExpr().getArray(e)) {
+			if (p.isOnline()) // double check
+				((Player) p).setBedSpawnLocation(loc, true);
```

Comment:
should `force` be `true`? the docs for the expression say "the bed still exists and is unobstructed", which could potentially be false if set manually. It looks like this was the existing behavior though so maybe it's juts a documentation fix that's needed?

Reply:
Yep, seems like it was [added ](https://github.com/SkriptLang/Skript/commit/d0e49bd57828f240cac19c6bd669bff102bfdf83#diff-0bea3bd93975380fd06a01d409b2e5bf4d1aba00bf11ec50c4b7f9e4da3a6bbfR69)the acceptChange method later without changing the description, will do it 👍